adapter/serviceregistry/aggregate/controller.go

```diff
@@ -79,23 +94,45 @@ func (c *Controller) ManagementPorts(addr string) model.PortList {
 // Instances retrieves instances for a service and its ports that match
 // any of the supplied labels. All instances match an empty label list.
 func (c *Controller) Instances(hostname string, ports []string,
-	labels model.LabelsCollection) []*model.ServiceInstance {
+	labels model.LabelsCollection) ([]*model.ServiceInstance, error) {
 	var instances []*model.ServiceInstance
+	var errs error
 	for _, r := range c.registries {
-		if instances = r.Instances(hostname, ports, labels); len(instances) > 0 {
-			break
+		var err error
+		instances, err = r.Instances(hostname, ports, labels)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		} else if len(instances) > 0 {
+			if errs != nil {
+				glog.Warningf("Instances() found match but encountered an error: %v", errs)
+			}
+			return instances, nil
 		}
 	}
-	return instances
+	return instances, errs
 }
 
 // HostInstances lists service instances for a given set of IPv4 addresses.
-func (c *Controller) HostInstances(addrs map[string]bool) []*model.ServiceInstance {
+func (c *Controller) HostInstances(addrs map[string]bool) ([]*model.ServiceInstance, error) {
 	instances := make([]*model.ServiceInstance, 0)
```

Comment:
maybe switch this to `out` so the variable names are clearer than `insts` and `instances`? same comment applies to the rest of the aggregate controller